### PR TITLE
OpenReg: support multiple executors

### PR DESCRIPTION
**Pull request by Zhenbin-8**
From PR https://github.com/pytorch/pytorch/pull/135646 we have split the daemon into drvier/executor, however, current executor stands for all devices and allocate memory all together. In order to better simulate device behavior, here we support multiple executors, each executor stands for one device.

cc @albanD @FFFrog

### CODE REVIEW

**Review by fffrog — APPROVED** [review]

**Review by albanD — COMMENTED** [review]
Looks good!
Just small typos!

File: test/cpp_extensions/open_registration_extension/pytorch_openreg/_aten_impl.py [hunk 1]
```diff
@@ -30,42 +30,34 @@ def _(*args, **kwargs):
...
-            args[1].stride(),
-        )
-    elif op is torch.ops.aten._local_scalar_dense.default:
-        args, _ = prepare_for_sending(args, {})
-        host_mem = driver.exec("send_data", *args)
-        return host_mem.item()
+    device = get_tensor_device(*args)
+    if device is None:
+        return _kernel_fallback(op, *args, **kwargs)
+    with DeviceContext(device):
```
[the remark this answers is not in the document]
Comment (Zhenbin-8) [reply test/cpp_extensions/open_registration_extension/pytorch_openreg/_aten_impl.py:55]: Sure, I'll fix it.

**Review by albanD — APPROVED** [review]
Thanks!

**Comment by Zhenbin-8** [comment]
@albanD Sorry to bother you again but I need your help merging this PR.

**Comment by albanD** [comment]
Ho yes this is expected, the CI needs to be triggered by a maintainer after changes are made. I'm on it!

**Pull request closed without merging**